objects/src/accounts/mod.rs

```diff
@@ -71,6 +71,27 @@ impl Account {
         Self { id, vault, storage, code, nonce }
     }
 
+    /// Creates and returns a new account constructed from the specified ID and delta.
+    pub fn from_delta(id: AccountId, delta: &AccountDelta) -> Result<Self, AccountError> {
```

Comment:
We should probably return an error from here if the delta is not for new account.